zrml/prediction-markets/src/lib.rs

```diff
@@ -2067,13 +2038,16 @@ mod pallet {
         impl_unreserve_bond!(unreserve_creation_bond, creation);
         impl_unreserve_bond!(unreserve_oracle_bond, oracle);
         impl_unreserve_bond!(unreserve_outsider_bond, outsider);
+        impl_unreserve_bond!(unreserve_dispute_bond, dispute);
         impl_slash_bond!(slash_creation_bond, creation);
         impl_slash_bond!(slash_oracle_bond, oracle);
         impl_slash_bond!(slash_outsider_bond, outsider);
+        impl_slash_bond!(slash_dispute_bond, dispute);
         impl_repatriate_bond!(repatriate_oracle_bond, oracle);
         impl_is_bond_pending!(is_creation_bond_pending, creation);
         impl_is_bond_pending!(is_oracle_bond_pending, oracle);
         impl_is_bond_pending!(is_outsider_bond_pending, outsider);
+        impl_is_bond_pending!(is_dispute_bond_pending, dispute);
 
         fn slash_pending_bonds(market_id: &MarketIdOf<T>, market: &MarketOf<T>) -> DispatchResult {
```

Comment:
```rust
            if Self::is_dispute_bond_pending(&market_id, &market, false) {
                Self::unreserve_dispute_bond(&market_id)?;
            }

            // Slash outstanding bonds; see
            // https://github.com/zeitgeistpm/runtime-audit-1/issues/34#issuecomment-1120187097 for
            // details.
            Self::slash_pending_bonds(&market_id, &market)?;
```

we unreserve the dispute instead of slashing it inside admin_destroy_market.

Reply:
5d6df7961b83d185a26b3e2b69a2f1f4b9005eb3